### PR TITLE
add autofix and push logic to trunk-action

### PR DESCRIPTION
**Pull request by jennygzma**
define INPUT_AUTOFIX_AND_PUSH if inputs.check-mode === payload.

if INPUT_AUTOFIX_AND_PUSH is defined, then run trunk check and push results to the github branch

handles unit test

### CODE REVIEW

File: .github/workflows/action_tests.yaml [hunk 1]
```diff
@@ -145,3 +145,61 @@ jobs:
...
+          git fetch origin
+          actual_commit_count=$(git rev-list --count feature-branch)
+          expected_commit_count=3
+          if [ "$actual_commit_count" -eq "$expected_commit_count" ]; then
+            echo "The branch has $expected_commit_count commits as expected."
+            exit 0
+          else
+            echo "The branch does not have $expected_commit_count commits."
+            exit 1
+          fi
```
Comment (sxlijin) [inline .github/workflows/action_tests.yaml:205]: let's base this assertion on the latest commit message for `origin/feature-branch`

**Comment by sxlijin** [comment]
otherwise looking good!

File: .github/workflows/action_tests.yaml [hunk 2]
```diff
@@ -194,12 +194,12 @@ jobs:
         run: |
           cd repo-under-test
           git fetch origin
-          actual_commit_count=$(git rev-list --count feature-branch)
-          expected_commit_count=3
-          if [ "$actual_commit_count" -eq "$expected_commit_count" ]; then
-            echo "The branch has $expected_commit_count commits as expected."
+          latest_commit_message=$(git log -1 --pretty=format:%s)
```
Comment (sxlijin) [inline .github/workflows/action_tests.yaml:198]:
```suggestion
          latest_commit_message=$(git log -1 --pretty=format:%s origin/feature-branch)
```

File: .github/workflows/action_tests.yaml [hunk 3]
```diff
@@ -194,12 +194,12 @@ jobs:
         run: |
           cd repo-under-test
```
Comment (sxlijin) [inline .github/workflows/action_tests.yaml:195]: should've mentioned this last time but: is it possible to do these assertions from src-repo?

File: .github/workflows/action_tests.yaml (hunk 2 above)
Comment (jennygzma) [reply .github/workflows/action_tests.yaml]: if i'm cd'd into src-repo i can just do `latest_commit_message=$(git log -1 --pretty=format:%s feature-branch)` right?

Comment (sxlijin) [reply .github/workflows/action_tests.yaml]: maybe?

the trickiness is that if you have a normal git repo at src-repo/, when you do `git push` from `repo-under-test/`. it'll add the commits from `repo-under-test/` to `src-repo/` but i'm not sure if it actually updates the working tree

Comment (sxlijin) [reply .github/workflows/action_tests.yaml]: `origin/feature-branch` will only work in `repo-under-test/`, not fom in `src-repo`

Comment (sxlijin) [reply .github/workflows/action_tests.yaml]: but you should try from `src-repo`

Comment (jennygzma) [reply .github/workflows/action_tests.yaml]: yeah i mean from src-repo git logs show that the autofix was committed to feature-branch on src-repo, so im assuming that it works - is that what u meant?

Comment (sxlijin) [reply .github/workflows/action_tests.yaml]: yeah, let's do the assertion from inside src-repo then

Comment (jennygzma) [reply .github/workflows/action_tests.yaml]: yah so this pr reflects that! unless this isn't wnat u meant?

Comment (sxlijin) [reply .github/workflows/action_tests.yaml]: oh, hm, when i made that comment the diff was still showing the repo-under-test assertions side of things. i must've not refreshed. mb.

**Review by sxlijin — APPROVED** [review]

**Pull request merged**